modules/web/src/app/shared/directives/value-cahnged.ts

```diff
@@ -0,0 +1,108 @@
+// Copyright 2023 The Kubermatic Kubernetes Platform contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Directive, ElementRef, OnInit} from '@angular/core';
+import {NgControl} from '@angular/forms';
+import {DialogModeService} from '@app/core/services/dialog-mode';
+
+@Directive({
+  selector: '[valueChanged]',
+})
+export class VlaueChangedDirective implements OnInit {
+  private _previousValue: any;
+
+  constructor(private _el: ElementRef, private _control: NgControl, private _isEditDialog: DialogModeService) {}
+
+  ngOnInit(): void {
+    if (!this._isEditDialog.isEditDialog) {
+      return;
+    }
+
+    const element = this._el.nativeElement;
+    this._previousValue = this._control.control.value;
+
+    switch (element.classList[0]) {
+      case 'mat-input-element':
+        if (element.getAttribute('type') === 'number') {
+          this._previousValue = !element.getAttribute('ng-reflect-model')
+            ? this._control.control.value
+            : parseInt(element.getAttribute('ng-reflect-model'));
+        }
+        break;
+      case 'mat-checkbox':
+        this._previousValue = this._control.control.value === null ? false : this._control.control.value;
+        break;
+      case 'mat-radio-group':
+        element.classList.toggle('radio-group-column', element.style.flexDirection === 'column');
```

Comment:
Why are we not saving initial value here?

Reply:
this class need to be add when `flexDirection` style is column so regardless the value

Reply:
Yes, I'm fine with the class but I'm asking why are we not saving `_previousValue` in this particular case?